### PR TITLE
As/fixes mega codereview

### PR DESCRIPTION
**Pull request by alsala**
Modifications for EthereumTransaction class after code review.
Added UT tests for serializations and semanticValidity

### CODE REVIEW

File: sdk/src/test/scala/com/horizen/api/http/SidechainJSONBOChecker.scala [hunk 1]
```diff
@@ -82,8 +82,9 @@ class SidechainJSONBOChecker {
     assertTrue(json.elements().asScala.length >= 3)
     assertTrue(json.get("id").isTextual)
     assertEquals(transaction.id, json.get("id").asText())
-    assertTrue(json.get("modifierTypeId").isNumber)
-    assertEquals(transaction.modifierTypeId.toInt, json.get("modifierTypeId").asInt())
+    // modifier type id is not part of json anymore
+    // assertTrue(json.get("modifierTypeId").isNumber)
+    //assertEquals(transaction.modifierTypeId.toInt, json.get("modifierTypeId").asInt())
```
Comment (i-Alex) [inline sdk/src/test/scala/com/horizen/api/http/SidechainJSONBOChecker.scala:87]: code can be removed completely.

File: sdk/src/test/java/com/horizen/account/transaction/EthereumTransactionSemanticValidityTest.java [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    }
+
+    @Test
+    public void testUnsignedTxValidity() {
+        var transaction = getUnsignedEoa2EoaEip1559Transaction();
+        assertNotValid(transaction);
+    }
+
+    @Test
+    public void testEoa2EoaEip1559Validity() {
```
[the remark this answers is not in the document]
Comment (alsala) [reply sdk/src/test/java/com/horizen/account/transaction/EthereumTransactionSemanticValidityTest.java:37]: I added some more UTs in the PR #601, and namely a testEoa2EosLegacyValidity with the missing cases.

File: sdk/src/main/java/com/horizen/account/transaction/EthereumTransaction.java [hunk 3]
```diff
@@ -195,14 +231,24 @@ public BigInteger getNonce() {
...
         return getMaxFeePerGas();
     }
 
+    @JsonProperty("gasPrice")
+    public BigInteger getJsonGasPrice() {
+        if (!this.isEIP1559())
+            return this.legacyTx().getGasPrice();
+        // for eip1559 tx this not an attribute of the object, it is computed using baseFee which depends on block height
+        return null;
+    }
```
[the remark this answers is not in the document]
Comment (alsala) [reply sdk/src/main/java/com/horizen/account/transaction/EthereumTransaction.java]: I could not find any annotation for doing that, therefore I left that in this way. I grouped them in a code section with a comment describing why they have been impelemnted.
I also made them private.

File: sdk/src/test/scala/com/horizen/api/http/SidechainJSONBOChecker.scala (hunk 1 above)
Comment (alsala) [reply sdk/src/test/scala/com/horizen/api/http/SidechainJSONBOChecker.scala]: Done

**Review by i-Alex — APPROVED** [review]

**Review by i-Alex — APPROVED** [review]

**Pull request merged**